src/lr_cd/lr_cd.py

```diff
@@ -0,0 +1,55 @@
+# lr_cd.py
+# author: Andy Zhang
+# date: 2024-01-10
+
+
+def coordinate_descent(X, y, w, alpha, ϵ=1e-4, max_iterations=1000):
+    """
+    Perform coordinate descent to minimize the mean square error of linear regression.
+
+    The function takes the predictor `X` and reponse `y` with initial guess for weights vector.
+    With an appropriate learning rate `alpha`, the function can return the optimized weights vector. 
+
+
+    Parameters
+    ----------
+    X : ndarray
+        Feature data matrix.
+    y : ndarray
+        Response data vector.
+    w : ndarray
+        Initial weights vector.
+    alpha : float
+        Learning rate.
+    ϵ : float, optional
+        Stop the algorithm if the change in weights is smaller than the value (default is 1e-4).
+    max_iterations : int, optional
+        Maximum number of iterations (default is 1000).
+
+    Returns
+    -------
+    ndarray
+        Optimized weights vector.
+
+
+
+    Examples
+    -------
+    >>> from lr_cd.lr_cd import coordinate_descent
+    >>> X = array([[ 1.        , -1.28162658],
```

Comment:
It seems like the data generation function only return a two columns dataframe, but here we required a 2D array for X, and 1D array for y

Reply:
I use np array here to match the style of sklearn, that is, X is a np array and y is another np array. See the new python scripts for more details.